app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -1191,4 +1218,11 @@ private void stopTimer() {
             }
         }
     }
+
+    public void onLocationChanged() {
+        boolean loaded = loadEntities();
+        if (!loaded) {
+            locationChangedWhileLoading = true;

```

Comment:
Otherwise, I really like the way `locationChangedWhileLoading` is enabling reloading of the entity list!